extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java

```diff
@@ -64,7 +60,8 @@ public void testMetrics() {
         assertEquals(0L, getCounterValueOrNull("hibernate-orm.entities.inserted"));
         Arc.container().requestContext().activate();
         try {
-            DummyEntity entity = new DummyEntity(12345L);
+            DummyEntity entity = new DummyEntity();
```

Comment:
Does that one throw the exception? It needs proxying?

Reply:
That code wasn't throwing an exception, as it was only storing a `DummyEntity`. If the test would have attempted to load it, you'd have had a runtime exception.

That seemed wrong to me (in many ways), so I made sure we throw an helpful (but fatal) error at build time to highlight the need for an empty constructor. Since it fails now, existing tests had to be fixed.

Reply:
(that specific test doesn't need proxying)